controllers/movieController.js

```diff
@@ -0,0 +1,146 @@
+            { model: models.Genres },
+        ],
+    })
+    response.send(movies)
+
+}
+async function movieById(request, response) {
+    var requestInfo = request.params.MovieID
+    const movies = await models.Movies.findAll({
+        where: {
+            id: requestInfo
+        },
+        attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+        include: [
+            { model: models.Directors },
+            { model: models.Genres },
+        ],
+    })
+    if (movies.length === 0) {
+        response.send('No Movie Found')
+    } else {
+        response.send(movies)
+    }
+}
+
+async function getByDirectorID(request, response) {
+    var requestInfo = request.params.directorId
+    const directorID = await models.Directors.findOne({
+        attributes: ['director'],
+        where: {
+            id: requestInfo
+        },
+        include: [{
+            model: models.Movies,
+            attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+            include: [{
+                model: models.Genres,
+                attributes: ['genre']
+            }]
+        }],
+    })
+    response.send(directorID)
+}
+async function getByGenre(request, response) {
+    var requestInfo = request.params.genre
+
+    const genre = await models.Genres.findOne({
+        attributes: ['genre'],
+        where: {
+            genre: requestInfo,
+        },
+        include: [{
+            model: models.Movies,
+            attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+            include: [{
+                model: models.Directors,
+                attributes: ['director']
+            }]
+        }]
+    })
+    response.send(genre)
+}
+async function postMovies(request, response) {
+    const { title, directors, releaseDate, rating, runTime, genres } = request.body
+
+    if (!title || !directors || !releaseDate || !rating || !runTime || !genres) {
+        response.status(400).send('the following are required title, directors, releaseDate, rating, runTime, genres if more then one director or genre seperate with comma')
+    }
+
+    const [movie] = await models.Movies.findOrCreate({
+        where: { title: title, releaseDate: releaseDate, runTime: runTime, ratings: rating },
+    })
+
+    await directors.split(',').forEach(async (name) => {
+        const [director] = await models.Directors.findOrCreate({
+            where: { director: name.trim() },
+        })
+        await models.MoviesDirector.findOrCreate({
+            where: { movieId: movie.id, directorId: director.id }
+        })
+    })
+
+    await genres.split(',').forEach(async (genre) => {
+        const [genres] = await models.Genres.findOrCreate({
+            where: { genre: genre.trim() },
+        })
+        await models.MovieGenre.findOrCreate({
+            where: { movieId: movie.id, genreId: genres.id }
+        })
+    })
+    response.send('Movie info posted')
```

Comment:
You really should return the full movie data to them especially since auto increment IDs are involved.